### PR TITLE
Update/dem

### PR DESCRIPTION
**Pull request by jmpearl**
# Summary

- This is a significant overhaul of the DEM package (may break scripts)
- It does the following:
  - adds solid boundary conditions to DEM package
    - infinite plane
    - finite planes (disk/rectangle)
    - finite cylinder
    - sphere and half sphere 
  - moves some DEM fieldLists from phys package to nodeList
  - simplifies node generation w/ DEM (no more nPerh) 
  - makes initialization on start up more robust
    - Hfield guaranteed to be appropriate
    - unique indices always valid
  - modified fast timestepping to allow variable neighbor-search buffer distances
  - fixes the "exploding" composite particle bug 

------
### ToDo :

- [x] Annotate ``RELEASE_NOTES.md`` with notable changes.
- [x] Create LLNLSpheral PR pointing at this branch. (PR#)
- [x] LLNLSpheral PR has passed all tests.

### CODE REVIEW

File: src/DEM/SolidBoundary/DEMBoundaryPolicyInline.hh
Comment (mdavis36) [inline src/DEM/SolidBoundary/DEMBoundaryPolicyInline.hh:1]: remove?

File: src/DEM/SolidBoundary/SolidBoundaryBase.hh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  virtual Vector distance(const Vector& position) const = 0;
+  virtual Vector velocity(const Vector& position) const = 0;
+
+  virtual void update(const double multiplier,
+                      const double t,
+                      const double dt) = 0;
+
+
+private:
+
```
Comment (mdavis36) [inline src/DEM/SolidBoundary/SolidBoundaryBase.hh:38]: empty scope

File: src/DEM/SolidBoundary/SolidBoundaryBaseInline.hh
Comment (mdavis36) [inline src/DEM/SolidBoundary/SolidBoundaryBaseInline.hh:1]: remove?

**Review by mdavis36 — CHANGES_REQUESTED** [review]
Just a few empty files and some nitpicking but otherwise looks good 👍

File: src/DEM/SolidBoundary/SolidBoundaryBaseInline.hh
Comment (jmpearl) [reply src/DEM/SolidBoundary/SolidBoundaryBaseInline.hh:1]: Thanks for catching these!

**Review by mdavis36 — APPROVED** [review]

**Pull request merged**